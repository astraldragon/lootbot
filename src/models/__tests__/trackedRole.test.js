const trackedRole = require("../trackedRole");

describe("models/trackedRole", () => {
  beforeEach(() => {
    this.model = {};

    this.sequelize = {
      define: jest.fn("define").mockReturnValue(this.model)
    };

    this.dataTypes = {
      STRING: "string",
      FLOAT: "float"
    };
  });

  it("should create the model definition", () => {
    trackedRole(this.sequelize, this.dataTypes);
    expect(this.sequelize.define).toMatchSnapshot();
  });
});
