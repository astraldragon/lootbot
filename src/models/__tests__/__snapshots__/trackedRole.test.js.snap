// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`models/trackedRole should create the model definition 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      "TrackedRole",
      Object {
        "guild": "string",
        "role": "string",
        "weight": "float",
      },
      Object {
        "paranoid": true,
      },
    ],
  ],
  "results": Array [
    Object {
      "isThrow": false,
      "value": Object {
        "associate": [Function],
      },
    },
  ],
}
`;
